crates/metaslang/cst/src/nodes.rs

```diff
@@ -6,6 +6,16 @@ use crate::cursor::{Cursor, CursorIterator};
 use crate::kinds::{KindTypes, NodeKind, TerminalKindExtensions};
 use crate::text_index::TextIndex;
 
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(transparent)]
+pub struct NodeId(usize);
+
+impl NodeId {
```

Comment:
not blocking: is it more idiomatic to use `From<>` and `Into<>` traits instead of this custom function?

Reply:
Yes, you're right. I was a bit skeptical because we're converting to `u32` in the WASM bindings, but I implemented the `TryFrom` trait and it just works.